### PR TITLE
Update parity-publish

### PR DESCRIPTION
**Pull request by Morganamilo**

### CODE REVIEW

File: Cargo.toml
Comment (Morganamilo) [reply Cargo.toml:833]: It's from the trigger CI commit. It will be removed once I confirm the check works as expected.

File: .github/workflows/check-semver.yml
Comment (lexnv) [inline .github/workflows/check-semver.yml:14]:
```suggestion
  TOOLCHAIN: nightly-2024-11-19
```
We can bump this higher, confirmed with https://github.com/paritytech/polkadot-sdk/actions/runs/11930690709/job/33252075574?pr=5732

File: Cargo.toml
Comment (lexnv) [reply Cargo.toml:833]: Check passed on https://github.com/paritytech/polkadot-sdk/pull/6535, should be good to go 🙏

File: .github/workflows/check-semver.yml
Comment (Morganamilo) [reply .github/workflows/check-semver.yml]: I was going off of the version the semver lib supports but yeah the newer nightly also works so will use that.

**Comment by Morganamilo** [comment]
The semver check is taking a really long time due to all the Cargo.toml changes and the new parity publish now checks those. This can probably be circumvented by having parity publish not bothering to compile the crates if no source changes and no dep changes. For now it shouldn't come up much.

**Review by EgorPopelyaev — APPROVED** [review]

**Comment by lexnv** [comment]
Would love to include this to origin/master to unblock semver-checks for:
- https://github.com/paritytech/polkadot-sdk/pull/5732

What are the next steps to merge this? 🙏

**Comment by Morganamilo** [comment]
There were still come issues with parity-publish .10.1. .10.2 is now released so merging as soon as approved.

**Review by bkontur — APPROVED** [review]

**Review by alvicsam — APPROVED** [review]

File: bridges/snowbridge/runtime/test-common/Cargo.toml
Comment (parity-crate-owner) [reply bridges/snowbridge/runtime/test-common/Cargo.toml:10]: Maybe the others aren't published? The changes were generated using the crates `parity-publish check` complained about.

**Review by lexnv — APPROVED** [review]
Nice job here! Thanks again for handling this 🙏

**Pull request merged**
